bigtable/admin/instance_admin.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_ADMIN_INSTANCE_ADMIN_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_ADMIN_INSTANCE_ADMIN_H_
+
+#include <memory>
+#include "bigtable/admin/instance_admin_client.h"
+#include "bigtable/client/internal/unary_rpc_utils.h"
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Implements a minimal API to administer Cloud Bigtable instances.
+ */
+class InstanceAdmin {
+ public:
+  /**
+   * @param client the interface to create grpc stubs, report errors, etc.
+   */
+  InstanceAdmin(std::shared_ptr<InstanceAdminClient> client)
```

Comment:
Do we expect a single InstanceAdminClient to be used by multiple InstanceAdmin?

Reply:
No.  And in a vacuum I would remove the `InstanceAdminClient` (now that you mention it, it did not occur to me before).  However, we do expect that instances of `bigtable::DataClient` will be shared by multiple `bigtable::Table` objects, and that instances of `bigtable::AdminClient` will be shared by different `bigtable::TableAdmin` objects.  Both will be rare use cases though.

Reply:
In that case can we pass in a unique_ptr with the InstanceAdminClient?

Reply:
I would prefer to keep it consistent with the other classes.  It would be surprising for users to learn that one (and the least common one at that) of the three client objects is used via `unique_ptr`.